agent/engine/docker_task_engine.go

```diff
@@ -1129,28 +1130,25 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	// and awsvpc. For reference - https://docs.docker.com/config/containers/logging/fluentd/.
 	if hostConfig.LogConfig.Type == logDriverTypeFirelens {
 		firelensContainers := task.GetFirelensContainers()
-		firelensVersion := firelensContainers[0].FirelensConfig.Version
+		firelensVersion := firelensContainers[0].GetFirelensVersion()
 		hostConfig.LogConfig = getFirelensLogConfig(task, container, firelensVersion, hostConfig, engine.cfg)
 
-		if firelensVersion != "v2" {
-			if task.IsNetworkModeAWSVPC() {
-				container.MergeEnvironmentVariables(map[string]string{
-					fluentNetworkHost: FluentAWSVPCHostValue,
-					fluentNetworkPort: FluentNetworkPortValue,
-				})
-			} else if container.GetNetworkModeFromHostConfig() == "" || container.GetNetworkModeFromHostConfig() == apitask.BridgeNetworkMode {
-				ipAddress, ok := getContainerHostIP(firelensContainers[0].GetNetworkSettings())
-				if !ok {
-					err := apierrors.DockerClientConfigError{Msg: "unable to get BridgeIP for task in bridge  mode"}
-					return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(&err)}
-				}
-				container.MergeEnvironmentVariables(map[string]string{
-					fluentNetworkHost: ipAddress,
-					fluentNetworkPort: FluentNetworkPortValue,
-				})
+		envVars := make(map[string]string)
+		if task.IsNetworkModeAWSVPC() {
+			getFirelensEnvironmentVariables(task, container, firelensVersion, envVars, AWSVPCHostValue)
```

Comment:
nit: This method is setting, not getting, maybe let's change the name to`setFirelensEnvironmentVariables`?

Reply:
Agreed and updated, Thanks!